### PR TITLE
feat: TraceQL metrics: max over time

### PR DESCRIPTION
**Pull request by javiermolinar**
**What this PR does**:
It implements the max_over_time aggregation function

**Checklist**
- [X] Tests updated
- [X] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/tempo/traceql/metrics-queries.md [hunk 1]
```diff
@@ -94,25 +97,37 @@ down by HTTP route.
...
 ```
 { name = "GET /:endpoint" } | count_over_time() by (span.http.status_code)
 
 ```
+
 The `min_over_time()` let you aggregate numerical values by computing the minimum value of them, such as the all important span duration.
 
 ```
-{ name = "GET /:endpoint" } | min_over_time(duration) by (span.http.target)
+{ name = "GET /:endpoint" } | min_over_time(duration` by (span.http.target)
```
Comment (knylander-grafana) [inline docs/sources/tempo/traceql/metrics-queries.md:112]: Is the back tick correct after `min_over_time(duration``? Should it be the closing parenthesis?

File: docs/sources/tempo/traceql/metrics-queries.md [hunk 2]
```diff
@@ -94,25 +97,37 @@ down by HTTP route.
...
 Any numerical attribute on the span is fair game.
 
 ```
 { name = "GET /:endpoint" } | min_over_time(span.http.status_code)
 ```
+
+The `max_over_time()` let you aggregate numerical values by computing the maximum value of them, such as the all important span duration.
+
+```
+{ name = "GET /:endpoint" } | max_over_time(duration` by (span.http.target)
```
Comment (knylander-grafana) [inline docs/sources/tempo/traceql/metrics-queries.md:124]: Same question here. It seems like the duration' should be duration), based on the other examples.

**Review by knylander-grafana — COMMENTED** [review]
Thank you for adding documentation! I have some questions about examples, but otherwise looks good.

**Comment by javiermolinar** [comment]
> Thank you for adding documentation! I have some questions about examples, but otherwise looks good.

You were right about the examples. Thank you for reviewing it!

File: docs/sources/tempo/traceql/metrics-queries.md [hunk 3]
```diff
@@ -59,6 +59,9 @@ These functions can be added as an operator at the end of any TraceQL query.
 `min_over_time`
 : Returns the minimum value of matching spans values per time interval (see the `step` API parameter)
 
+`max_over_time`
+: Returns the minimum value of matching spans values per time interval (see the `step` API parameter)
```
Comment (mdisibio) [inline docs/sources/tempo/traceql/metrics-queries.md:63]:
```suggestion
: Returns the maximum value of matching spans values per time interval (see the `step` API parameter)
```

**Review by mdisibio — APPROVED** [review]
LGTM - We got all the hard changes out of the way with min_over_time, and this is straightforward. 👍

**Pull request merged**